sdk/core/Azure.Core/src/Shared/RequestContentHelper.cs

```diff
@@ -24,6 +26,12 @@ internal static class RequestContentHelper
             return content;
         }
 
+#if !NET5_0_OR_GREATER
+        // RequiresUnreferencedCode in net5.0 doesn't have AttributeTargets.Class as a target, but it was added in net6.0
+        // The call to JsonSerializer.Serialize is not compatible with trimming, but is only used in NET 5 and below, so don't warn otherwise.
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCode(SerializationTrimmingMessage)]
+        [System.Diagnostics.CodeAnalysis.RequiresDynamicCode(SerializationTrimmingMessage)]
```

Comment:
Yeah I was seeing a couple warnings (2026 and 3050), I was surprised when these actually worked to make them go away

Reply:
I figured it out. The file `RequestContentHelper.cs` is compiled into `Azure.Monitor.OpenTelemetry.Exporter` which only builds for `netstandard2.0`, it doesn't target `net6.0`. So it uses the "old" code which produces the warning.

To fix this, we need to target that project to `net6.0` as well.

@TimothyMothra FYI

Reply:
I was already expecting this change and have it in my branch already.
If you need to make this change in this PR, that's fine with me :)

Reply:
I think it's OK to leave the warning in - remove these attributes in this file (as they should not be there really). There are other warnings from the AOT test which originate in the OTel Exporter, this one just doesn't look like it because it's from the shared code - but it does originate there as well.

I would leave this one up to Timothy to resolve.

Reply:
Sounds good! Thanks for letting me know - I will remove it here